CONTRIBUTING.md

```diff
@@ -0,0 +1,13 @@
+# Contributing
+
+## Release a new version
+
+All commit messages have to follow the [conventionalcommits.org](https://conventionalcommits.org) format specification.
+
+1. Make sure your are on the `master` branch.
+2. Make sure that you have committed everything and that your code is working.
+3. Run `npx standard-version`. This command updates the changelog with description of changes to your code since the last released version. It then increments the code version. All these changes are commited into the `master` branch.
+4. Push these changes to the remote repository and its `master` branch.
```

Comment:
I'm unsure about the part that says all changes should be committed and pushed to the `master` branch, considering that we rely on contributing via PRs.

I think that for releasing a new version, we should _minimize_ direct changes to the `master` branch and instead, prefer something like:
1. `git pull` the most recent changes.
2. Run `npx standard-version`.
3. Push these changes to the remote `master` branch.

*OR*, for even more safety, the 3rd step could be:
3. Create a new PR with the release.

Then,
4. A Github Action is triggered […]

WDYT? @zuzanawangle @prokopsimek

Reply:
Understood. When I was preparing this process, I assumed that only BCMS staff would be doing the versioning, so it was mentioned that they should do the versioning on the main branch. It is not a problem to create a new version in PR, but there may be a problem in the whole process with the automatic release. If the release commit will be in a separate branch, then I don't think the auto-release on the commit from the master branch will work because it won't have the correct format for Github Action.

Reply:
> When I was preparing this process, I assumed that only BCMS staff would be doing the versioning, so it was mentioned that they should do the versioning on the main branch

I'm 99% sure this is the correct assumption, nevertheless, I feel that doing any major changes through the PR is usually a safer option than doing them through the main branch directly, even if only a few people would have permission to do that.

> but there may be a problem in the whole process with the automatic release. If the release commit will be in a separate branch, then I don't think the auto-release on the commit from the master branch will work because it won't have the correct format for Github Action.

Hmm, what if we merge the release PR w/o squashing? That would push the release commit into the main branch, which should trigger the auto-release, I believe. The flow would be:
1. Create a release PR.
2. Reviewer approves.
3. Merge the release PR w/o squashing.
4. After the merge, GH Action is triggered and does an auto-release.

Or am I wrong?

---

Either way, I emphasize that I suggested a PR as an optional alternative:
> OR, for even more safety, the 3rd step could be:
> 3. Create a new PR with the release.

If there are any major issues with doing this through the PR, then I'm fine w/ pushing to the main branch directly. I was mostly concerned about steps and overall tone that seemed to encourage pushing commits other than release-related to the main branch directly.

Reply:
> Hmm, what if we merge the release PR w/o squashing? That would push the release commit into the main branch, which should trigger the auto-release, I believe. The flow would be:

The original process set originally in this PR is the following:
1. Create PR with some changes
2. Review PR
3. Merge PR to master branch (doesn't matter if squashed or not). The important thing is that the BCMS staff now sets the merge commit message to something meaningful and conventional commits valid to possibly generate a changelog automatically with standard-version tool afterward (I don't expect that all the contributors will use conventional commits when contributing).
4. BCMS staff checkouts to master branch
5. Runs the `npx standard-version` command. There will be auto-generated changelog based on conventional commits.
6. BCMS staff pushes changes to master branch and the commit now has `chore(release): xyz` format
7. It triggers the Github Action and the package is automatically published to npmjs.com


We can also modify what the standard-version does so it doesn't have to commit all the changes and can give a space to update the changelog first but BCMS staff has to commit the changes generated by the library with the correct commit name to activate the Github action correctly.

Reply:
> The original process set originally in this PR is the following:
> 
> 1. Create PR with some changes
> 2. Review PR
> 3. Merge PR to master branch (doesn't matter if squashed or not). The important thing is that the BCMS staff now sets the merge commit message to something meaningful and conventional commits valid to possibly generate a changelog automatically with standard-version tool afterward (I don't expect that all the contributors will use conventional commits when contributing).
> 4. BCMS staff checkouts to master branch
> 5. Runs the `npx standard-version` command. There will be auto-generated changelog based on conventional commits.
> 6. BCMS staff pushes changes to master branch and the commit now has `chore(release): xyz` format
> 7. It triggers the Github Action and the package is automatically published to npmjs.com

Sounds good to me. Let's just modify the CONTRIBUTING.md to better reflect that.